packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts

```diff
@@ -0,0 +1,511 @@
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { IGCRuntimeOptions, ISummarizer } from "@fluidframework/container-runtime";
+import {
+    requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+    ITestObjectProvider,
+    createSummarizer,
+    summarizeNow,
+    waitForContainerConnection,
+    mockConfigProvider,
+    ITestContainerConfig,
+} from "@fluidframework/test-utils";
+import { describeNoCompat, ITestDataObject, itExpects } from "@fluidframework/test-version-utils";
+import { delay, stringToBuffer } from "@fluidframework/common-utils";
+import { IContainer, LoaderHeader } from "@fluidframework/container-definitions";
+import { IOdspResolvedUrl } from "@fluidframework/odsp-driver-definitions";
+import { getUrlFromItemId, MockDetachedBlobStorage } from "../mockDetachedBlobStorage";
+
+/**
+ * These tests validate that SweepReady attachment blobs are correctly marked as tombstones. Tombstones should be added
+ * to the summary and changing them (sending / receiving ops, loading, etc.) is not allowed.
+ */
+describeNoCompat("GC attachment blob tombstone tests", (getTestObjectProvider) => {
+    const sweepTimeoutMs = 200;
+    const settings = {};
+    const gcOptions: IGCRuntimeOptions = { inactiveTimeoutMs: 0 };
+    const testContainerConfig: ITestContainerConfig = {
+        runtimeOptions: {
+            summaryOptions: {
+                summaryConfigOverrides: {
+                    state: "disabled",
+                },
+            },
+            gcOptions,
+        },
+        loaderProps: { configProvider: mockConfigProvider(settings) },
+    };
+
+    let provider: ITestObjectProvider;
+    let mainContainer: IContainer;
+    let mainDataStore: ITestDataObject;
+
+    async function loadContainer(summaryVersion: string) {
+        return provider.loadTestContainer(
+            testContainerConfig,
+            { [LoaderHeader.version]: summaryVersion },
+        );
+    }
+
+    describe("Attachment blobs in attached container", () => {
+        let summarizer: ISummarizer;
+
+        beforeEach(async function() {
+            provider = getTestObjectProvider({ syncSummarizer: true });
+            if (provider.driver.type !== "local") {
+                this.skip();
+            }
+
+            settings["Fluid.GarbageCollection.ThrowOnTombstoneUsage"] = true;
+            settings["Fluid.GarbageCollection.TestOverride.SweepTimeoutMs"] = sweepTimeoutMs;
+        });
+
+        async function createMainContainerAndSummarizer() {
+            mainContainer = await provider.makeTestContainer(testContainerConfig);
+            mainDataStore = await requestFluidObject<ITestDataObject>(mainContainer, "default");
+
+            // Send an op to transition the container to write mode.
+            mainDataStore._root.set("transition to write", "true");
+            await waitForContainerConnection(mainContainer);
+
+            summarizer = await createSummarizer(
```

Comment:
I'll leave it up to you:
I personally prefer to just return the value of `await createSummarizer` and then set the variable in the test below, instead of having to scroll up and find the summarizer variable. There may be a little more redundancy, but it's easier to trace the code.

Reply:
I don't mind either way. Updated it based on your suggestion.